crates/cxx-qt-gen/src/parser/parameter.rs

```diff
@@ -16,29 +20,48 @@ pub struct ParsedFunctionParameter {
 }
 
 impl ParsedFunctionParameter {
+    /// This function parses the list of arguments
     pub fn parse_all_without_receiver(
         signature: &Signature,
     ) -> Result<Vec<ParsedFunctionParameter>> {
-        signature
-            .inputs
-            .iter()
-            .map(|input| {
-                match input {
-                    FnArg::Typed(type_pattern) => {
-                        let parameter = ParsedFunctionParameter::parse(type_pattern)?;
-
-                        // Ignore self as a parameter
-                        if parameter.ident == "self" {
-                            return Ok(None);
-                        }
-                        Ok(Some(parameter))
-                    }
+        let mut iter = signature.inputs.iter();
+
+        let missing_self_arg = "First argument must be a supported `self` receiver type!\nUse `&self` or `self: Pin<&mut Self>` instead.";
+        match iter.next() {
+            Some(FnArg::Typed(type_pattern)) => {
+                let parameter = ParsedFunctionParameter::parse(type_pattern)?;
+                if parameter.ident == "self" && types::is_pin_of_self(&parameter.ty) {
+                    Ok(())
+                } else {
+                    Err(Error::new_spanned(type_pattern, missing_self_arg))
+                }
+            }
+            Some(FnArg::Receiver(Receiver {
+                reference: Some(_), // `self` needs to be by-ref, by-value is not supported.
+                mutability: None,   // Mutable self references are not supported.
+                ..
+            })) => Ok(()), /*Okay, found a &self reference*/
```

Comment:
why is this `/* ... */` whereas the others are `//` ?

Reply:
Done